### PR TITLE
CBMC version 6 release process changes

### PR DESCRIPTION
**Pull request by NlightNFotis**
This PR contains (temporary) changes while we're orchestrating the new (v6) release of CBMC going forward.

For now, it deactivates pushing build artefacts to third party services, and increments the version number.

### CODE REVIEW

File: CMakeLists.txt
Comment (esteffin) [inline CMakeLists.txt:44]: This section is to avoid a problem with CMake.

Essentially CMake accepts version of type `major[.minor[.patch[.tweak]]]` and fails if anything else is fed.

Before we were passing anything specified in `config.inc` as version and leaving CMake to validate it.

In this way we are just passing the part of the version that is composed of digits and `.`s, so `6.0.0-preview` is considered a valid version and for CMake it is interpreted as `6.0.0` allowing CBMC itself to use the full version specified in `config.inc`.

**Review by esteffin — APPROVED** [review]

File: .github/workflows/release-packages.yaml
Comment (thomasspriggs) [inline .github/workflows/release-packages.yaml:152]: :pick: It might be good to put a short comment on each of these disabled sections to explain why it is disabled. It is possible that someone might see that they are not receiving new versions from Docker / homebrew and go looking here. So some comments could save some confusion.

**Review by peterschrammel — APPROVED** [review]

File: src/config.inc
Comment (thomasspriggs) [inline src/config.inc:79]: I think the major version also needs to be propagated to here - https://github.com/diffblue/cbmc/blob/eaa2cdd997249de233aeacfdef7d3adf55cb7c2c/src/goto-programs/write_goto_binary.h#L15C1-L15C2

File: v6_undo_stack.markdown
Comment (thomasspriggs) [inline v6_undo_stack.markdown:15]: Commit hashes could be really handy here, assuming that we can avoid invalidating them in a re-base before the PR is merged.

**Review by thomasspriggs — APPROVED** [review]

File: src/config.inc
Comment (kroening) [reply src/config.inc:79]: No, please don't tweak that without a reason. It's coincidental that CBMC is at version 5.x.x and that the goto binary version is 5.

**Comment by kroening** [comment]
Please wait with merging until we've got an agreed plan for version 6.

File: src/config.inc
Comment (thomasspriggs) [reply src/config.inc:79]: I'd heard it said that they were supposed to coincide. Thank you for clarifying.

**Comment by thomasspriggs** [comment]
> Please wait with merging until we've got an agreed plan for version 6.

@kroening I have now outlined a plan for the process to use for the major release on the Version 6 RFC here - https://github.com/diffblue/cbmc/issues/7743#issuecomment-1789552377 I have outlined it on the RFC rather than here, so that the plan along with the discussion around it is straightforward to reference even once PR(s) are merged / closed.

**Comment by NlightNFotis** [comment]
Hello @kroening, may I request another look at this one please?

We're keen to move with the implementation, as we're in a relatively tight timeline with version 6, so if there are no blockers, we want to move with this one and sort out any minor disagreements in follow-up PRs.

**Comment by martin-cs** [comment]
> We're keen to move with the implementation, as we're in a relatively tight timeline with version 6

Is the time-line and plan for version 6 available anywhere?  There are a bunch of interface-breaking changes it would be good to make to `goto-analyzer` and `goto-instrument` as part of it.

**Comment by TGWDB** [comment]
> > We're keen to move with the implementation, as we're in a relatively tight timeline with version 6
> 
> Is the time-line and plan for version 6 available anywhere? There are a bunch of interface-breaking changes it would be good to make to `goto-analyzer` and `goto-instrument` as part of it.

As raised (a long time ago) in #7743 we'd hoped to be we've been planning this for some time. The exact details of which Version 6 labelled PRs are merged will depend on if they're ready. The actual timeline is ASAP, I'd suggest raising the PR soon if you can. We'd love to start merging PRs and have unstable/pre-releases from... now. (In practice we don't want to cause too much collateral and so have not merged anything breaking yet.)

**Comment by NlightNFotis** [comment]
Hello, I've merged this one now in the interest of moving the stack of dependent PRs forward (along with the rest of the process).

Feel free to leave any more comments here, or in https://github.com/diffblue/cbmc/issues/7743

This is just the foundation of the rest of the work, so if you find anything objectionable here after the merge, feel free to ask for it and we can raise a new PR to cover that.

File: .github/workflows/release-packages.yaml
Comment (NlightNFotis) [reply .github/workflows/release-packages.yaml:152]: I will consider this for a next PR, but I want to move this one forward to unblock us.

In the meantime, I'm hoping that someone coming across this change would see the `blame` long which would lead him to the commit https://github.com/diffblue/cbmc/pull/7987/commits/2d14e9996f197f4373d8438a300713a66307ebfe and its message that explains the situation.

File: v6_undo_stack.markdown
Comment (NlightNFotis) [reply v6_undo_stack.markdown:15]: I'm not sure if we need commit hashes here - this was supposed to be a casual thing just to verify our reverted changes against the log here, sort of a double-entry bookkeeping practice but for the purpose of managing the process.

But if keen, I can also add this in a new PR.

**Pull request merged**
